include/modem/location.h

```diff
@@ -0,0 +1,299 @@
+	uint16_t timeout;
+
+	/** Used cellular positioning service. */
+	enum loc_service service;
+};
+
+/** WiFi positioning configuration. */
+struct loc_wifi_config {
+	/** Timeout (in seconds) on how long WiFi positioning procedure can take. */
+	uint16_t timeout;
+
+	/** Used WiFi positioning service. */
+	enum loc_service service;
+};
+
+/** Positioning method configuration. */
+struct loc_method_config {
+	/** Positioning method. */
+	enum loc_method method;
+	union {
+		/** Configuration for LOC_METHOD_CELLULAR. */
+		struct loc_cellular_config cellular;
+		/** Configuration for LOC_METHOD_GNSS. */
+		struct loc_gnss_config gnss;
+		/** Configuration for LOC_METHOD_WIFI. */
+		struct loc_wifi_config wifi;
+	};
+};
+
+/** Location request configuration. */
+struct loc_config {
+	/** Number of location methods in 'methods'. */
+	uint8_t methods_count;
+	/**
+	 * @brief Selected positioning methods and associated configurations in priority order.
+	 *
+	 * @details Index 0 has the highest priority. Number of methods is indicated in
+	 * 'methods_count'.
+	 */
+	struct loc_method_config *methods;
+	/**
+	 * @brief Position update interval in seconds.
+	 *
+	 * @details Set to 0 for a single position update. For periodic position updates
+	 * the valid range is 10...65535 seconds.
+	 */
+	uint16_t interval;
+};
+
+/**
+ * @brief Event handler prototype.
+ *
+ * @param[in] event_data Event data.
+ */
+typedef void (*location_event_handler_t)(const struct loc_event_data *event_data);
+
+/** @brief Initializes the library.
+ *
+ * @details Initializes the library and sets the event handler function.
+ *
+ * @param[in] event_handler Event handler function.
+ *
+ * @return 0 on success, or negative error code on failure.
+ */
+int location_init(location_event_handler_t event_handler);
+
+/**
+ * @brief Requests the current position or starts periodic position updates.
+ *
+ * @details Requests the current position using the given configuration. Depending on the
+ *          configuration, a single position or periodic position updates are given. The results are
+ *          delivered to the event handler function.
+ *
+ *          Periodic position updates can be stopped by calling location_cancel().
+ *
+ * @param[in] config Used configuration or NULL to get a single position update with
+ *                   the default configuration.
+ *
+ * @return 0 on success, or negative error code on failure.
+ */
+int location_request(const struct loc_config *config);
+
+/**
+ * @brief Cancels periodic position updates.
+ *
+ * @return 0 on success, or negative error code on failure.
+ */
+int location_request_cancel(void);
+
+/**
+ * @brief Sets default values to given configuration.
+ *
+ * @details Given methods are set as list of methods for the given configuration.
+ * Caller is responsible for allocating and deallocating memory for both configuration and
+ * method list. 'methods' memory area must be kept allocated as long as the configuration is used.
```

Comment:
```suggestion
 * @details Given methods are set as a list of methods for the given configuration.
 * Caller is responsible for allocating and deallocating memory for both configuration and
 * method list. The 'methods' memory area must be kept allocated as long as the configuration is used.
```

Reply:
@tokangas, Depends on who adds the comment :) I'll align these.